doc/custom-widgets.md

```diff
@@ -0,0 +1,139 @@
+
+```ts
+import { Component, Inject, PLATFORM_ID } from '@angular/core';
+import { BaseWidget, NgAisInstance } from 'angular-instantsearch';
+import { connectMenu } from 'instantsearch.js/es/connectors';
+
+@Component({
+  selector: 'ng-ais-menu-select',
+  template: '<p>It works!</p>'
+})
+export class MenuSelect extends BaseWidget {
+  constructor(
+    @Inject(PLATFORM_ID) public platformId: Object,
+    searchInstance: NgAisInstance
+  ) {
+    super(searchInstance, 'MenuSelect');
+  }
+
+  public ngOnInit() {
+    this.createWidget(connectMenu, { attributeName: 'categories' });
+    super.ngOnInit();
+  }
+}
+```
+
+And voilà, your widget is connected to InstantSearch.js and the state of the search it self!
+
+### 3. Render from the state
+
+Your component instance has access a property `this.state` which holds the rendering options of the widget.
+
+From the [connectMenu](https://community.algolia.com/instantsearch.js/v2/connectors/connectMenu.html) InstantSearch.js connector documentation, we can see that we have access to:
+
+* `items[]` -> The elements that can be refined for the current search results.
+* `createURL(item.value)` -> Creates the URL for a single item name in the list.
+* `refine(item.value)` -> Filter the search to item value.
+* `canRefine` -> True if refinement can be applied.
+* `isShowingMore` -> True if the menu is displaying all the menu items.
+* `toggleShowMore` -> Toggles the number of values displayed between `limit` and `showMoreLimit`.
+* `canToggleShowMore` -> `true` if the toggleShowMore button can be activated (enough items to display more or already displaying more than limit items).
+
+For instance if you want to access the items, you will need to use `this.state.items` into your component code.
+
+Let's write together the component template:
+
+```ts
+import { Component, Inject, PLATFORM_ID } from '@angular/core';
+import { BaseWidget, NgAisInstance } from 'angular-instantsearch';
+import { connectMenu } from 'instantsearch.js/es/connectors';
+
+@Component({
+  selector: 'ng-ais-menu-select',
+  template: `
+    <select
+      class="menu-select"
+      (change)="state.refine($event.target.value)"
+    >
+      <option
+        *ngFor="let item of state.items"
+        [value]="item.value"
+        [selected]="item.isRefined"
+      >
+        {{item.label}}
+      </option>
+    </select>
+  `
+})
+export class MenuSelect extends BaseWidget {
+  constructor(
+    @Inject(PLATFORM_ID) public platformId: Object,
+    searchInstance: NgAisInstance
+  ) {
+    super(searchInstance, 'MenuSelect');
+  }
+
+  public ngOnInit() {
+    this.createWidget(connectMenu, { attributeName: 'categories' });
+    super.ngOnInit();
+  }
+}
+```
+
+You can now use your directive `<ng-ais-menu-select></ng-ais-menu-select>` to display your custom MenuSelect widget!
+
+See your freshly crafted MenuSelect in action [here](http://angular-instantsearch.netlify.com/dev-novel/?selectedStory=CustomWidgets.MenuSelect).
```

Comment:
We have a live example of this newly created MenuSelect in our [widget showcase].